src/scenes/levelend-scene/levelend-scene.ts

```diff
@@ -27,28 +25,23 @@ export class LevelEndScene {
   public audioPlayer: AudioPlayer;
   public isLastLevel: boolean;
   public levelEndElement = document.getElementById('levelEnd');
+  public nextButtonInstance: NextButtonHtml;
+  public retryButtonInstance: RetryButtonHtml;
+  public mapButtonInstance: MapButton;
   public riveMonster: RiveMonsterComponent;
   public canvasElement: HTMLCanvasElement;
-  constructor(
-    height: number,
-    width: number,
-    starCount: number,
-    currentLevel: number,
-    switchToGameplayCB,
-    switchToLevelSelectionCB,
-    data,
-  ) {
-    this.height = height;
-    this.width = width;
+  constructor(switchToGameplayCB, switchToLevelSelectionCB) {
+    const {starCount, currentLevel, data} = gameStateService.getLevelEndSceneData();
     this.switchToGameplayCB = switchToGameplayCB;
     this.switchToLevelSelectionCB = switchToLevelSelectionCB;
     this.data = data;
     this.audioPlayer = new AudioPlayer();
     this.canvasElement = document.getElementById("rivecanvas") as HTMLCanvasElement;
     this.starCount = starCount;
     this.currentLevel = currentLevel;
-    this.isLastLevel = this.currentLevel ===
-    this.data.levels[this.data.levels.length - 1].levelMeta.levelNumber;
+    this.isLastLevel =
+      this.currentLevel ===
+      this.data.levels[this.data.levels.length - 1].levelMeta.levelNumber;
```

Comment:
This should be computed from `gameStateService`

Reply:
done here. please check @miguelccodev